ibis/backends/snowflake/converter.py

```diff
@@ -10,3 +19,23 @@ def convert_JSON(s, dtype, pandas_type):
         return s.map(converter, na_action="ignore").astype("object")
 
     convert_Struct = convert_Array = convert_Map = convert_JSON
+
+
+class SnowflakePyArrowData(PyArrowData):
+    @classmethod
+    def convert_table(cls, table: pa.Table, schema: Schema) -> pa.Table:
+        import pyarrow as pa
+
+        columns = [cls.convert_column(table[name], typ) for name, typ in schema.items()]
+        return pa.Table.from_arrays(columns, names=schema.names)
+
+    @classmethod
+    def convert_column(cls, column: pa.Array, dtype: dt.DataType) -> pa.Array:
+        if dtype.is_json() or dtype.is_array() or dtype.is_map() or dtype.is_struct():
+            import pyarrow as pa
+
+            if isinstance(column, pa.ChunkedArray):
+                column = column.combine_chunks()
```

Comment:
I don't really like this, but `ExtensionArray` apparently does not allow `ChunkedArray`s to be passed to `from_storage`

Reply:
I'm not a fan either, but realistically this is what would happen upstream.